### ISSUE TITLE
Moving the Frictionless community from Discord to Slack

### ISSUE THREAD

**Issue by sapetti9** [frictionlessdata/frictionlessdata.io#716]
We are moving the Frictionless community chat from Discord to Slack (with a Matrix mirror). 

Please join us: https://join.slack.com/t/frictionlessdata/shared_invite/zt-14x9bxnkm-2y~uQcmmrqarSP2kV39_Kg

## Tasks

* [x] Open Slack channel
* [ ] Import message history 
* [x] Build Matrix bridge (@aivuk)

**Comment by roll** [comment]
@lwinfree 
@sapetti9 
@aivuk 
If we still experience problems with Matrix bridge next week before the planning I would suggest creating a separate issue for it and putting it to "Priority" to get more info/duscussion before investing more time into it. Currently it blocks Edgar from other work

**Comment by aivuk** [comment]
@roll I think it will be solved until next Monday. I just finished configuring the delegation and I was able to access the bridged rooms from different matrix servers and solved also the room permissions that we were having on it. The only missing thing that I'm working now is to use one puppet for all users that wants to use Matrix. Right now the puppet is working only for my user, making it possible to other matrix users to access the rooms but they are not able to post messages. I'm trying a last solution now and I will update the issue today

**Comment by aivuk** [comment]
Ok, we decided to host it on https://element.io/matrix-services and the service is in process to be acquired. But let me explain a brief description of what happened so far:

 1. First we decided to test the integration and I proposed to use my server at vaz.io.
 2. I registered the domain on my server matrix.vaz.io.
 3. Configure Matrix Synapse server there and configured the SSL certificates for it. Tested it with @sapetti9 and it worked as a Matrix server. Next step was trying the Matrix-Slack bridge.
 4. I then installed on my server the node.js application https://github.com/matrix-org/matrix-appservice-slack in my server and created an old version of the slack api token key to use with it. 
 5. After struggling a bit to make the integration work, finally we had some test channels from Frictionless Slack bridged in their own rooms in Matrix. Then we discovered two problems: I. we received messages from Slack in Matrix, but we couldn't send messages from Matrix to Slack; II. after we request a test from the user Augusto, we discovered that users from other matrix servers (like from @user:matrix.org instead of my server @user:matrix.vaz.io) couldn't not access the server. 
 6. To fix II I did need to configure the federation in my server (https://federationtester.matrix.org/#matrix.vaz.io), including configuring the firewall to open the port 8448. This fixed  the issue allowing users from other server.
 7. Couldn't find the problem that was blocking other users to post from Matrix to Slack and decided to test this other solution https://matrix.org/docs/projects/bridge/matrix-puppet-slack. This worked but I found out that this was doing an integration only for my user on slack/matrix, and it didn't do it automatically for all the users.
 8. We decided to pay https://etke.cc/ to configure the server and the slack/matrix bridge. For that we did need to have a server already configured to provide access to them.
 9. After some discussion with the tech team we decided to use the cheapest virtual machine from google cloud. I did need one day to setup, configure the dns poiting matrix.okfn.org to it, there were some mistakes done on cloudfare dns. 
 10. I saw that the OS installed on the google cloud server had a old version not recommended by the Matrix team and I did need to update all the python dependencies and install synapse directly from the project, not using the not supported package from the server.
 11. Finally after the Matrix was properly configured on okfn.org server including the federation (you can see it here https://federationtester.matrix.org/#matrix.okfn.org), the next step would be to contact etke.cc and ask them to configure the integration. But after a meeting with @amercader, he suggested to not waste more time on the server and that it would give us probably unexpected headaches maintaining the matrix server and it would be better at least for now to pay https://element.io/matrix-services
 12. I contacted Element sales team and had a response today where they explained three important points to us: 1. they can provide us a dump of their database if we decide to migrate to our own server; 2. the solution to matrix slack bridge is the same open source one that I was trying to install, this means that we can use it in the future; 3. the users that they count in the price are just for users in our own domain @user:matrix.okfn.org, but most of our users are going to be from external servers, we are not going to create a open registration server in our instance.
 13. Finally we are right now in the process of signing the service and configuring it. As soon as it done I will update the issue.

**Comment by aivuk** [comment]
On Friday I signup for Element Cloud service and configured the Matrix server for OKFN. It is accessible on https://matrix.okfn.org and the Slack integration is finally configured. There were some problems with it, I was not able to add any room to the integration, even after configuring it:

![slack-integration-problem](https://user-images.githubusercontent.com/272892/166104016-4617a192-5433-4cb6-b059-3023bbe4062a.png)


I contacted their support team but I found out a way to fix it today!

Now it is  configured and tested for all the current Slack channels (#beginners, #datasets, #frictionless-planet, #general, #random). On Matrix I created a public FrictionlessData Space and we can start sharing the address:

https://matrix.to/#/#frictionlessdata:matrix.okfn.org

And this will bring an user from Matrix to the space where she can choose to join one of the channels. It is possible also to share an invite for each individual room inside the Matrix client interface.
